### PR TITLE
Animation improvements for dApp info window

### LINKED ISSUE #2524: Improvements in Dapp connection info animations
Make the animations in Daap connection info window more friendly for users.  

**To do:**
- make a more friendly transition when the user disconnected the app (take a look at the gif)
CleanShot 2022-10-28 at 16 36 37

- improve Dapp connection info window opening view (take a look at the gif)
98742393-176f108a-e212-4b0c-b8c9-4fa78fa541ed

- fix the issue for the transition when the user closed the accordion. The transition should be smooth as below. Make sure the tooltip for the default wallet is displayed correctly.
198961665-1e75c62e-bc26-44cf-a2b8-6ebb2aa917f8

### PR DESCRIPTION
**Pull request by kkosiorowska**
Closes #2524 

This PR makes the animations for dApp info window more friendly for users.

**UI**

https://user-images.githubusercontent.com/23117945/200262240-32f044c5-065f-4a47-a1e2-86813f94ed2d.mov

https://user-images.githubusercontent.com/23117945/200260276-7bc506c3-cc8a-45a9-b683-df5f4a89ab0a.mov

https://user-images.githubusercontent.com/23117945/200260287-c47f66c4-cd95-4490-a1fd-fb076274a31b.mov

**To Test**
- [ ] check how the transition looks when the user opened the window
- [ ] check how the transition looks when the user disconnected the app  
- [ ] check how the transition looks when the user closed/opened the accordion. Make sure the tooltip for the default wallet is displayed correctly.

Latest build: [extension-builds-2562](https://github.com/tallycash/extension/suites/9183682159/artifacts/428097822) (as of Tue, 08 Nov 2022 08:32:26 GMT).

### CODE REVIEW

File: ui/components/Shared/SharedAccordion.tsx [hunk 1]
```diff
@@ -17,14 +19,37 @@ export default function SharedAccordion({
...
+    setIsOpen((open) => !open)
+  }
 
   useEffect(() => {
     setHeight(contentHeight ?? contentRef.current?.clientHeight ?? 600)
   }, [contentRef, contentElement, contentHeight])
 
+  useEffect(() => {
+    if (isOpen) {
+      setTimeout(() => {
```
Comment (jagodarybacka) [inline ui/components/Shared/SharedAccordion.tsx:43]: if you are using timeouts then you have to make sure you destroy them before the component is unmounted

**Review by jagodarybacka — CHANGES_REQUESTED** [review]

**Review by greg-nagy — APPROVED** [review]
code looks good, works well

waiting for a @VladUXUI sign off before merge

File: ui/components/Shared/SharedAccordion.tsx [hunk 2]
```diff
@@ -17,14 +19,35 @@ export default function SharedAccordion({
...
   const [isOpen, setIsOpen] = useState(isInitiallyOpen)
+  /**
+   * Accordion content has an overflow hidden property when it is closed.
+   * The overflow property is changed to visible after a delay when the content is visible.
+   * The goal is to display the tooltip message correctly.
+   */
+  const [isVisible, setIsVisible] = useState(isInitiallyOpen)
+  /* If the accordion is open by default, the first opening should be without a transition. */
+  const [withTransition, setWithTransition] = useState(!isInitiallyOpen)
   const [height, setHeight] = useState(0)
```
Comment (hyphenized) [inline ui/components/Shared/SharedAccordion.tsx:30]: We probably want to initialize this to `contentHeight` if it's available and `isInitiallyOpen` is truthy

**Review by hyphenized — APPROVED** [review]
Looks much better, nice work!

**Review by jagodarybacka — APPROVED** [review]
Looks very nice ✨

**Pull request merged**